include/boost/geometry/algorithms/detail/distance/geometry_to_segment_or_box.hpp

```diff
@@ -220,7 +220,8 @@ class geometry_to_segment_or_box
         comparable_return_type cd_min1(0);
         point_iterator_type pit_min;
         seg_or_box_iterator_type it_min1 = seg_or_box_points.begin();
-        seg_or_box_iterator_type it_min2 = ++seg_or_box_points.begin();
+        seg_or_box_iterator_type it_min2 = it_min1;
+        ++it_min2;
```

Comment:
So I think that your proposal should still fail on Solaris. It could be:

    seg_or_box_iterator_type it_min1 = boost::const_begin(seg_or_box_points);
    seg_or_box_iterator_type it_min2 = it_min2;
    ++it_min2;

or

    seg_or_box_iterator_type it_min1 = boost::const_begin(seg_or_box_points);
    seg_or_box_iterator_type it_min2 = boost::prior(boost::const_begin(seg_or_box_points));

I prefer the first one because it's one dependency less, less functions to debug, probably faster for some iterators which musn't be created two times, boost::prior() call doesn't look intuitive enough for me etc. But I can be persuaded otherwise :)